controller/volume_controller.go

```diff
@@ -1831,8 +1831,13 @@ func (vc *VolumeController) replenishReplicas(v *longhorn.Volume, e *longhorn.En
 				log.Debugf("Failed replica %v will be reused during rebuilding", reusableFailedReplica.Name)
 				reusableFailedReplica.Spec.FailedAt = ""
 				reusableFailedReplica.Spec.HealthyAt = ""
-				reusableFailedReplica.Spec.RebuildRetryCount++
-				vc.backoff.Next(reusableFailedReplica.Name, time.Now())
+
+				replicaRebuildStatus := reusableFailedReplica.Status.RebuildStatus
+				if replicaRebuildStatus == nil || replicaRebuildStatus.FailedReason != longhorn.ReplicaRebuildFailedReasonDisconnection || replicaRebuildStatus.RebuildDisconnectCount >= longhorn.ReplicaRebuildDisconnectCountMax {
+					reusableFailedReplica.Spec.RebuildRetryCount++
+					vc.backoff.Next(reusableFailedReplica.Name, time.Now())
+				}
+
```

Comment:
The essential of this enhancement is, a failed replica on a down node should not get stuck in the reuse retry loop and even reach the max retry limitation. 
To handle it, can we simply refresh the backoff window when the node is still down and the previous backoff window is passed, rather than using another counter `RebuildDisconnectCount`?